tests/test_repository.py

```diff
@@ -37,37 +55,49 @@ def test_gpg_signature_structure_is_preserved():
         "gpg_signature": ("filename.asc", "filecontent"),
     }
 
-    tuples = repository.Repository._convert_data_to_list_of_tuples(data)
+    tuples = repository.Repository._convert_metadata_to_list_of_tuples(data)
     assert tuples == [("gpg_signature", ("filename.asc", "filecontent"))]
 
 
-def test_content_structure_is_preserved():
-    """Preserve 'content' key when converting metadata."""
-    data = {
-        "content": ("filename", "filecontent"),
-    }
-
-    tuples = repository.Repository._convert_data_to_list_of_tuples(data)
-    assert tuples == [("content", ("filename", "filecontent"))]
-
-
 def test_iterables_are_flattened():
     """Flatten iterable metadata to list of tuples."""
     data = {
         "platform": ["UNKNOWN"],
     }
 
-    tuples = repository.Repository._convert_data_to_list_of_tuples(data)
+    tuples = repository.Repository._convert_metadata_to_list_of_tuples(data)
     assert tuples == [("platform", "UNKNOWN")]
 
     data = {
         "platform": ["UNKNOWN", "ANOTHERPLATFORM"],
     }
 
-    tuples = repository.Repository._convert_data_to_list_of_tuples(data)
+    tuples = repository.Repository._convert_metadata_to_list_of_tuples(data)
     assert tuples == [("platform", "UNKNOWN"), ("platform", "ANOTHERPLATFORM")]
 
 
+def test_form_metadata_translation(example_distribution):
+    data = example_distribution.metadata_dictionary()
+    tuples = repository.Repository._convert_metadata_to_list_of_tuples(data)
+    for key, value in tuples:
+        assert isinstance(key, str)
+        assert isinstance(value, str) or value is None
+
+
+def test_form_metadata_translation_everything(monkeypatch):
```

Comment:
And this is a variation on the above test which again, doesn't seem as helpful.

Reply:
I'll remove this too.

Reply:
I think there is value in keeping this test. I added some comments explaining what the test does. It should be clearer now.